src/prefect/engine/result.py

```diff
@@ -123,6 +123,23 @@ def store_safe_value(self) -> None:
                 value=value, result_handler=self.result_handler
             )
 
+    def validate(self) -> bool:
+        """
+        Run any validator functions associated with this result and return whether the result is valid or not.
+
+        Returns:
+            - bool: whether or not the Result passed all validation functions
+        """
+        if self.run_validators and self.validators:
+            for validation_fn in self.validators:
+                is_valid = validation_fn(self)
```

Comment:
Considered taking *args and **kwargs all up in here, in case we pass in context for them (or users want to call validate themselves, passing in more random stuff to their functions) but I think we expect the validation functions to be parameterizable only by the actual data they need to validate; thoughts?

In fact I'm passing the whole `Result`, though really I think we expect people's functions to be mostly dependent on Result.value. I'm considering unpacking Result.value on their behalf for convenience, like `validation_fn(self.value, self)`; thoughts?

Reply:
Yeah I believe they will only need to be parameterizable by the data needed for validation.

They will _most likely_ be dependent on the `Result.value` but I do like having the whole result object in case needed. Throwing in my 2𝇍 but I don't feel strongly for it

Reply:
I agree @joshmeek , if they are providing their own functions at Result creation, they can throw most variable data onto that result as they need it (also makes it reproducible). It isn't clear where a user would provide extra variable info during the TaskRunner pipeline.

@lauralorenz  I like the `validation_fn(self)` approach.

Reply:
Roger that team, leaving it as is then